hyperopt/spark.py

```diff
@@ -377,6 +384,33 @@ def run_dispatcher():
                 cur_time = timeit.default_timer()
                 elapsed_time = cur_time - start_time
 
+                # check early stopping condition if it is defined
+                if self.early_stop_fn:
+                    curr_num_ok_trials = 0
+                    for trial in self.trials:
+                        if trial["result"]["status"] == STATUS_OK:
+                            curr_num_ok_trials += 1
+
+                    if curr_num_ok_trials > prev_num_ok_trials:
+                        # check early stopping condition
+                        stop, kwargs = self.early_stop_fn(
+                            self.trials, *self.early_stop_args
+                        )
+                        self.early_stop_args = kwargs
+
+                        if stop:
+                            # same logic as timeouts
+                            self.trials._fmin_cancelled = True
+                            self.trials._fmin_cancelled_reason = (
+                                "early stopping condition"
+                            )
+                            self._cancel_running_trials()
+                            logger.warning(
+                                "fmin cancelled because of "
+                                + self.trials._fmin_cancelled_reason
+                            )
```

Comment:
Could you make the part to be a function ? then in timeout case and early stop case we can reuse the function.